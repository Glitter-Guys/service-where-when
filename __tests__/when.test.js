import React from 'react';
import expect from 'expect';
import When from './../client/src/components/when.jsx';


describe('<When />', () => {
  const dummyWhenData = {
    startTime: '2018-03-13T02:00:00Z',
    endTime: '2018-03-13T04:00:00Z',
    series: 'weekly',
  };
  const whenWrap = shallow(<When whenData={dummyWhenData} />);

 it('should be defined', () => {
   expect(When).toBeDefined();
 });

 it('should render a When component', () => {
   expect(whenWrap).toMatchSnapshot();
 });

 it('should display a single day event as date and time', () => {
   const whenDateClassEl = whenWrap.find('.when__date');
   expect(whenDateClassEl).toHaveLength(1);
 });

 it('should display a multiday event as start and end date-time', () => {
  const dummyWhenDataWithMultiDay = {
     startTime: '2018-03-10T18:00:00Z',
     endTime: '2018-03-12T00:00:00Z',
     series: null,
     multiDay: true,
  };
  const whenWrapWithMulti = shallow(<When whenData={dummyWhenDataWithMultiDay} />);
  const whenStartClassEl = whenWrapWithMulti.find('.when__start');
  expect(whenStartClassEl).toHaveLength(1);
 });

 it('should display series info', () => {
    const seriesTxt = whenWrap.find('[data-test="when__series"]').text();
    expect(seriesTxt).toBe('Repeats every week');
  });

  it('should not display series info when null', () => {
    const dummyWhenDataWithNull = {
      startTime: '2018-03-13T02:00:00Z',
      endTime: '2018-03-13T04:00:00Z',
      series: null,
    };
    const whenWrapWithNull = shallow(<When whenData={dummyWhenDataWithNull} />);
    const seriesTxt2 = whenWrapWithNull.find('[data-test="when__series"]').text();
    expect(seriesTxt2).toBe('');
   });

});
