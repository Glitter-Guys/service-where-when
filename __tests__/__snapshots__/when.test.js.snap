// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<When /> should render a When component 1`] = `
<div
  className="textSection"
>
  <div
    className="icon"
  >
    <InlineSVG
      cacheGetRequests={false}
      height="24"
      onLoad={[Function]}
      src="./icons/clock.svg"
      supportTest={[Function]}
      uniquifyIDs={true}
      viewBox="0 0 24 24"
      width="24"
      wrapper={[Function]}
    />
  </div>
  <div
    className="when__text"
  >
    <span>
      <span
        className="when__date"
      >
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element="time"
          filter={[Function]}
          format="dddd, MMMM D YYYY"
          fromNow={false}
          interval={60000}
          onChange={[Function]}
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
        >
          2018-03-13T02:00:00Z
        </t>
      </span>
      <span
        className="when__time"
      >
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element="time"
          filter={[Function]}
          format="h:mm A"
          fromNow={false}
          interval={60000}
          onChange={[Function]}
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
        >
          2018-03-13T02:00:00Z
        </t>
         to 
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element="time"
          filter={[Function]}
          format="h:mm A"
          fromNow={false}
          interval={60000}
          onChange={[Function]}
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
        >
          2018-03-13T04:00:00Z
        </t>
      </span>
    </span>
    <span
      className="secondaryText"
      data-test="when__series"
    >
      Repeats every week
    </span>
  </div>
</div>
`;
